onnxruntime 1.22.2

tombenes opened the pull request:
onnxruntime 1.22.2

**Destination channel:** Defaults

### Links

- [PKG-8643]
- dev_url:        https://github.com/microsoft/onnxruntime/tree/v1.22.2
- conda_forge:    https://github.com/conda-forge/onnxruntime-feedstock
- pypi:           https://pypi.org/project/onnxruntime/1.22.2
- pypi inspector: https://inspector.pypi.io/project/onnxruntime/1.22.2/

### Explanation of changes:

- new version number
- aligned with conda-forge multi-output format
- aligned version naming with conda-forge
- disabled testing for cuda variants (local tests on cuda enabled machine are passing on both linux and windows)


[PKG-8643]: https://anaconda.atlassian.net/browse/PKG-8643?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

tombenes committed:
1.22.0

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_acn9zbc1pz/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:0: no_global_test: Global tests are ignored in multi-output recipes.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:58: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:59: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
===== Final Report: =====
7 Errors and 11 Warnings were found

</details>

tombenes committed:
fix linux-aarch64

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_16yer140qe/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:59: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:58: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
add os-specific args

tombenes committed:
change order of variants for faster failures

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_1fn_kpce_5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:59: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:58: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
enable glibc 2.28

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_edk5piqd8_/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:58: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:59: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:86: outputs_not_unique: output "onnxruntime": Output name is not unique
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
temp skip cpu builds

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_79tkuuj7bf/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:88: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:88: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
temp build only for py310

tombenes committed:
disable all patches

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_7a18op35tx/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
debug find nvcc

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_397y_h80oj/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
add nvcc to host

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_4ecryeyoqs/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
regenerate patch for windows

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_86hkwhkt_s/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
build all variants

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_b5ghcb2hw2/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
build cuda only

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_c73vcw5_n5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:89: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
===== Final Report: =====
5 Errors and 10 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_87_lh8s46l/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:79: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:94: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:94: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_2c55qreo6t/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:68: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:79: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:94: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:94: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_ddp2ggra7c/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_287bbky552/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:65: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:77: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:81: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:96: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:96: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_4exida4542/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip failing tests on linux

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_9aonhonlml/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
skip testing cuda variants

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_a2g503uuqt/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:63: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:62: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:64: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:66: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:61: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:60: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime": Output name is not unique
- clone/recipe/meta.yaml:95: outputs_not_unique: output "onnxruntime-novec": Output name is not unique
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime": The requirements section is missing.
- clone/recipe/meta.yaml:0: missing_section: output "onnxruntime-novec": The requirements section is missing.
===== Final Report: =====
5 Errors and 11 Warnings were found

</details>

tombenes committed:
add host deps for -cpp; pin host deps

tombenes committed:
fix pinnings

tombenes committed:
skip testing cuda variants

tombenes committed:
enable all variants

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_68u3ug85yl/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

tombenes committed:
skip testing cuda on windows

tombenes committed:
disable unnecessary patches

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_06zzsuqht1/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

tombenes committed:
whitelist windows system and built libs; patches cleanup

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_76glebdb5l/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:1: missing_license_file: The recipe is missing the `about/license_file` or `about/license_url` key.
- clone/recipe/meta.yaml:1: missing_description: The recipe is missing a description

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:0: missing_section: The about section is missing.
- clone/recipe/meta.yaml:1: missing_documentation: The recipe is missing a doc_url or doc_source_url
- clone/recipe/meta.yaml:0: patch_unnecessary: An unexpected error occurred. Please report this issue to the pi-automation team through the #pi-automation channel.
- clone/recipe/meta.yaml:1: missing_home: The recipe is missing a homepage URL
- clone/recipe/meta.yaml:1: missing_summary: The recipe is missing a summary
- clone/recipe/meta.yaml:1: missing_package_name: The recipe is missing a package name
- clone/recipe/meta.yaml:1: missing_package_version: The recipe is missing a package version
- clone/recipe/meta.yaml:1: missing_license: The recipe is missing the `about/license` key.
- clone/recipe/meta.yaml:0: missing_section: The package section is missing.
- clone/recipe/meta.yaml:1: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:0: missing_tests: No tests were found.
- clone/recipe/meta.yaml:0: missing_section: The build section is missing.
- clone/recipe/meta.yaml:0: missing_section: The requirements section is missing.
- clone/recipe/meta.yaml:1: missing_build_number: The recipe is missing a build number
===== Final Report: =====
15 Errors and 2 Warnings were found

</details>

tombenes committed:
restore meta.yaml

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_19vrtv80sz/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

tombenes committed:
update to 1.22.2

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_91l8tp26zd/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

tombenes committed:
fix windows build

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_191pk5lgot/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

tombenes committed:
rename dll on windows

tombenes committed:
fix patch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_fb9dvn5iwn/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-novec-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime-cpp": No tests were found.
- clone/recipe/meta.yaml:1: missing_tests: output "onnxruntime": No tests were found.
===== Final Report: =====
4 Errors and 0 Warnings were found

</details>

ifitchet commented on recipe/build.sh:
```diff
@@ -2,34 +2,36 @@
 
 set -exuo pipefail
 
+find ${PREFIX} -name nvcc
```
Debug?

tombenes replied:
Ugh, yeah. Doesn't hurt the package tho, I think.

ifitchet commented on recipe/build.sh:
```diff
@@ -2,34 +2,36 @@
-if [[ "${target_platform:-other}" == 'osx-arm64' ]]; then
-    OSX_ARCH="arm64"
+if [[ "$(uname -s)" == "Linux" ]]; then
+    OS_SPECIFIC_ARGS="--allow_running_as_root"
```
Is this for Docker?

tombenes replied:
Yup, specifically the CI, idk how it works on the new build platform but prefect was running docker containers as root

ifitchet commented on recipe/meta.yaml:
```diff
@@ -73,19 +79,68 @@ requirements:
+      commands:                  # [(ep_variant == "cpu")]
+        - test -f $PREFIX/include/onnxruntime/core/session/onnxruntime_cxx_api.h  # [unix and (ep_variant == "cpu")]
+        - test -f $PREFIX/lib/libonnxruntime${SHLIB_EXT}                          # [unix and (ep_variant == "cpu")]
+        # - test -f $PREFIX/lib/libonnxruntime_providers_shared${SHLIB_EXT}  # [unix and (ep_variant == "cuda")]
+        # - test -f $PREFIX/lib/libonnxruntime_providers_cuda${SHLIB_EXT}    # [unix and (ep_variant == "cuda")]
```
You said above that the tests passed on a CUDA-enabled Docker instance.  So, presumably, these tests should work on such a system.  For non-CUDA-enabled systems they's not be used because of the selector.

Same for Windows, below?

tombenes replied:
Yes, I've ran both linux docker (with `--gpus all`) and windows. I updated PR description to state that explicitly.

tombenes replied:
The comment-out on those lines actually comes from our instances being unable to install the test env, as they can't find the `__cuda` virtual package. The real tests are run by the build script

ifitchet commented on recipe/meta.yaml:
```diff
@@ -1,65 +1,71 @@
     - ninja     # [not win]
-    - patch     # [unix and (ep_variant == "cuda")]
-    - m2-patch  # [win]
+    - psutil
```
`psutil` in `build` and `host`?

tombenes replied:
Hmm yeah that probably shouldn't have ended up here

cbouss commented:
I haven't had time to verify, but I strongly suggest to close/reopen the PR, so that the build gets triggered on pbp at least once before merge.

JamesRobertsonGames commented on abs.yaml:
```diff
@@ -1,2 +1,5 @@
+# Required for glibc >= 2.28
+pkg_build_image_tag: main-rockylinux-8
+
 build_env_vars:
-  ANACONDA_ROCKET_ENABLE_PY313 : yes
+  ANACONDA_ROCKET_GLIBC: "2.28"
```
Unnecessary on new build system (this is the default)

ifitchet replied:
I _do_ have to enable that in a Docker instance, though.  🤷

tombenes replied:
Leaving it for now just in case

ifitchet replied:
Although, that might have been cured by a `git pull` for the latest `cbc.yaml` which has no further mention of `ANACONDA_ROCKET_GLIBC`.  It could be me being a klutz!

tombenes commented:
> I haven't had time to verify, but I strongly suggest to close/reopen the PR, so that the build gets triggered on pbp at least once before merge.

I'll do that in one commit addressing all the review feedback. Considering the 48h build time, I don't want to trigger them multiple times.

JamesRobertsonGames commented on recipe/meta.yaml:
```diff
@@ -1,65 +1,71 @@
+    - setuptools
     - wheel
+    - flake8 7.1.1
+    - libdate 3.0.1
     - packaging
-    - python-flatbuffers
+    - python-flatbuffers 24.3.25
+    - optional-lite 3.5.0
```
Where are these very specific version requirements coming from?

tombenes replied:
Those are compiled host deps, linter was complaining about pinning them.

JamesRobertsonGames reviewed (commented):
Minor comments on top of Ian's comments - do run on pbp to see if you're getting similar builds. It's unlikely but it may differ

tombenes committed:
review fixes

JamesRobertsonGames reviewed (approved):
Still require one more review on this but happy with where it is at 👍

ifitchet reviewed (approved):
If only we could get it to build!

tombenes committed:
output ctest logs

tombenes committed:
use clang-14 on osx

tombenes committed:
limit windows builds to 4 threads

tombenes committed:
skip fastmath test on aarch64

tombenes committed:
skip QDQTransformerTests

tombenes committed:
skip non-deterministic tests

tombenes committed:
skip non-deterministic tests

tombenes committed:
skip all fastmath tests

ifitchet commented:
Hmm, just perusing a completed _build_ and saw:
```
2025-09-24 11:01:32,575 build [ERROR] - No module named 'onnx'
Traceback (most recent call last):
  File "/home/task_175870986121774/conda-bld/onnxruntime_1758710357088/work/tools/ci_build/build.py", line 1690, in run_onnxruntime_tests
    import onnx  # noqa: F401
ModuleNotFoundError: No module named 'onnx'
2025-09-24 11:01:32,576 build [WARNING] - onnx is not installed. The ONNX tests will be skipped.
2025-09-24 11:01:32,576 build [WARNING] - onnxmltools and keras are not installed. The keras tests will be skipped.
```

but I guess that's covered by [requirements-dev.txt](https://github.com/microsoft/onnxruntime/blob/v1.22.2/requirements-dev.txt#L6) and just a distraction.

tombenes commented:
> Hmm, just perusing a completed _build_ and saw:
> 
> ```
> 2025-09-24 11:01:32,575 build [ERROR] - No module named 'onnx'
> Traceback (most recent call last):
>   File "/home/task_175870986121774/conda-bld/onnxruntime_1758710357088/work/tools/ci_build/build.py", line 1690, in run_onnxruntime_tests
>     import onnx  # noqa: F401
> ModuleNotFoundError: No module named 'onnx'
> 2025-09-24 11:01:32,576 build [WARNING] - onnx is not installed. The ONNX tests will be skipped.
> 2025-09-24 11:01:32,576 build [WARNING] - onnxmltools and keras are not installed. The keras tests will be skipped.
> ```
> 
> but I guess that's covered by [requirements-dev.txt](https://github.com/microsoft/onnxruntime/blob/v1.22.2/requirements-dev.txt#L6) and just a distraction.

`onnx` is vendored and linked statically (the cpp part). If we were to run those tests (by including our own `onnx` with python) we'd need to unvendor `onnx` and a lot of other stuff (big one `protobuf`), otherwise the build system gets confused.
